File: backend/Models/Book.cs
﻿using System;
using System.Collections.Generic;

namespace backend.Models;
// book model (class), defines all the properties and their data types
public partial class Book
{
    public int BookId { get; set; }

    public string Title { get; set; } = null!;

    public string Author { get; set; } = null!;

    public string Publisher { get; set; } = null!;

    public string Isbn { get; set; } = null!;

    public string Classification { get; set; } = null!;

    public string Category { get; set; } = null!;

    public int PageCount { get; set; }

    public double Price { get; set; }
}

File: frontend/src/types/Book.ts
// book interface, essentially just an object, defining all the keys and their data types
export interface Book {
    bookId: number,
    title: string,
    author: string,
    publisher: string,
    isbn: string,
    classification: string, 
    category: string,
    pageCount: number, 
    price: number,
}